base_images/fedora_base-setup.sh

```diff
@@ -57,8 +57,9 @@ if ! ((CONTAINER)); then
         # vs google-network-daemon.service.  Fix this with a custom
         # cloud-init service file.
         CLOUD_SERVICE_PATH="systemd/system/cloud-init.service"
-        echo "$sourcemsg" > /etc/$CLOUD_SERVICE_PATH
-        cat $SCRIPT_DIRPATH/fedora-cloud-init.service >> /etc/$CLOUD_SERVICE_PATH
+        echo "$sourcemsg" | $SUDO tee -a /etc/$CLOUD_SERVICE_PATH
```

Comment:
This is not equivalent: the old code clobbers `$CLOUD_SERVICE_PATH`, the new code appends. If I were a future maintenance programmer, I would see the `-a` and immediately start looking for when/where/who creates and initializes that file. I would encourage you to drop the `-a` here and in the other instance below.

And, taking a step back, I don't understand the reason for the `tee`. Is it so the contents are visible in a log somewhere? If so, it doesn't seem very helpful, because it's just undescribed content. Maybe an 'echo this is what $CLOUD_SERVICE_PATH will look like'?

Reply:
> This is not equivalent

Oh woops, good catch.

> I don't understand the reason for the tee

The previous build process had root access to run the script, the new build process *sometimes* has root access.  The `$SUDO tee` guarantees the file can always be created.